### ISSUE TITLE
NoExtensionFunctionOnNullableReceiver clarification?

### ISSUE THREAD

**Issue by ursusursus** [Faire/faire-detekt-rules#129]
Hi,
NoExtensionFunctionOnNullableReceiver flags this
```
fun Foo?.toBar(): Bar? {
    ...
}
```

but doesn't flag this

```
fun Foo?.toBar(): Bar {
    ...
}
```

i.e works based on return type being nullable or not

was this desired? I'd best like to ban all extensions over nullable types
